### DISCUSSION TITLE
How do I add my blog posts to the sitemap?

### DISCUSSION THREAD

**Discussion by arnab** [avitorio/outstatic#221] [General]
First, Outstatic looks great, and exactly what I am looking for, to add to my website. Kudos, @avitorio !

How do I go about adding the published posts into my `sitemap`? I already have some pages and have configured [Next.js to programatically generate it](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-a-sitemap-using-code-js-ts).

**Comment by avitorio** [comment]
Hey @arnab,
Thanks for the kind words and I'm glad you are enjoying Outstatic! 

You could use a similar logic as the one in your `generateStaticParams()` function to get the paths you want to add to your sitemap and generate one at build time. I never did this but should be a good contribution to our advanced-blog template if you ever get around to it. Find more info on the [Next.js sitemap docs](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-a-sitemap-using-code-js-ts).

```javascript
import { BASE_URL } from '@/app/lib/constants'
 
export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
  const db = await load()
  const items = await db
    .find(
      {
        collection: 'posts',
        status: 'published'
      },
      ['slug', 'publishedAt']
    )
    .toArray()

  
  return items.map((item) => ({
    url: `${BASE_URL}/${item.slug}`,
    lastModified: item.publishedAt,
  }))
}
``` 

Let me know if you manage to get it working. Cheers!

**Reply by arnab** [reply]
Sounds good, I will try this out.

_(And happy to add this to https://github.com/avitorio/outstatic/tree/canary/examples/advanced-blog/ as an example once I get it working.)_

**Reply by arnab** [reply]
That worked well! Thanks @avitorio !
